### PR TITLE
roles: do not limit docker_memory_limit for various daemons

### PR DESCRIPTION
**Pull request by neha-ojha**
Since we do not have enough data to put valid upper bounds for the memory
usage of these daemons, do not put artificial limits by default. This will
help us avoid failures like OOM kills due to low default values.

Whenever required, these limits can be manually enforced by the user.

More details in
https://bugzilla.redhat.com/show_bug.cgi?id=1638148

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1638148
Signed-off-by: Neha Ojha <nojha@redhat.com>

### CODE REVIEW

**Comment by neha-ojha** [comment]
running ./generate_group_vars_sample.sh results in this https://paste.fedoraproject.org/paste/u57z9q9IP0Nr2X0JYtaDkg, not sure we want that.

**Comment by leseb** [comment]
@neha-ojha yes it is, sorry it seems that an earlier PR forgot to resync. Could you please add one more commit to your PR?

1. One tag re-sync group_vars with master, typically before "roles: do not limit docker_memory_limit for various daemons" commit
2. And then the one that adds your changes?

Thanks.

**Comment by guits** [comment]
see #3264

**Comment by leseb** [comment]
Ok forget about my comment then :)

**Comment by guits** [comment]
@neha-ojha I rebased your branch on master

**Comment by leseb** [comment]
jenkins test this please

**Comment by leseb** [comment]
jenkins test pipeline

**Comment by neha-ojha** [comment]
@guits @leseb thanks for taking care of the rebase, can you help me understand the failures in the jenkins pipeline. I do not see anything related.

**Comment by leseb** [comment]
jenkins test pipeline

**Comment by guits** [comment]
the failure is unrelated

File: roles/ceph-iscsi-gw/defaults/main.yml [hunk 1]
```diff
@@ -80,13 +80,13 @@ trusted_ip_list: 192.168.122.1
 # These options can be passed using the 'ceph_mds_docker_extra_env' variable.
 
 # TCMU_RUNNER resource limitation
-ceph_tcmu_runner_docker_memory_limit: 1g
+ceph_tcmu_runner_docker_memory_limit: "{{ ansible_memtotal_mb }}m"
 ceph_tcmu_runner_docker_cpu_limit: 1
 
 # RBD_TARGET_GW resource limitation
-ceph_rbd_target_gw_docker_memory_limit: 1g
+ceph_rbd_target_gw_docker_memory_limit: "{{ ansible_memtotal_mb }}"
```
Comment (leseb) [inline roles/ceph-iscsi-gw/defaults/main.yml]: +m?

File: roles/ceph-iscsi-gw/defaults/main.yml [hunk 2]
```diff
@@ -80,13 +80,13 @@ trusted_ip_list: 192.168.122.1
...
 ceph_rbd_target_gw_docker_cpu_limit: 1
 
 # RBD_TARGET_API resource limitation
-ceph_rbd_target_api_docker_memory_limit: 1g
+ceph_rbd_target_api_docker_memory_limit: "{{ ansible_memtotal_mb }}"
```
Comment (leseb) [inline roles/ceph-iscsi-gw/defaults/main.yml]: +m

**Comment by guits** [comment]
@neha-ojha the failure in `ceph-ansible-prs-dev-lvm_batch` is unrelated

**Comment by leseb** [comment]
jenkins test luminous-docker_cluster

**Pull request merged**